core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -107,4 +121,26 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
 
     return newFiles;
   }
+
+  /**
+   * Traverses the history of the table's current snapshot and finds the snapshot with the given snapshot id as its
+   * parent.
+   * @return null if the passed in snapshot is not present in the table, else the snapshot for which the given snapshot
+   * is the parent
+   * @throws IllegalArgumentException when the given SnapshotId is not found in the table
+   * @throws IllegalStateException when the given snapshot ID is not an ancestor of the current table state
+   */
+  public static Snapshot snapshotAfter(Table table, long snapshotId) {
+    Snapshot previousSnapshot = table.snapshot(snapshotId);
```

Comment:
Minor: You may want to rename this `parent` so that it is clear why the loop exits.

Reply:
Actually, I think it may be a good idea to change the loop to exit early. That would be a bit cleaner. And there's no need to keep the parent around at all, or to check that it is non-null every time through the loop.

```
Preconditions.checkArgument(table.snapshot(snapshotId) != null, ...);

Snapshot current = table.currentSnapshot();
while (current != null) {
  if (current.parentId() == snapshotId) {
    return current;
  }
  current  = table.snapshot(current.parentId());
}

throw new IllegalStateException(...);
```